test/staking/SwapHelper.test.ts

```diff
@@ -0,0 +1,403 @@
+    console.log("Deployed goodfund manager", {
+      manager: goodFundManager.address,
+    });
+
+    console.log("deployed contribution, deploying reserve...", {
+      founder: founder.address,
+    });
+
+    console.log("setting permissions...");
+    const tokenUsdOracleFactory = await ethers.getContractFactory(
+      "BatUSDMockOracle"
+    );
+    setDAOAddress("UNISWAP_ROUTER", uniswapRouter.address);
+
+    daiUsdOracle = await tokenUsdOracleFactory.deploy();
+    const compUsdOracleFactory = await ethers.getContractFactory(
+      "CompUSDMockOracle"
+    );
+    compUsdOracle = await compUsdOracleFactory.deploy();
+    goodCompoundStaking = await deployStaking(
+      dai.address,
+      cDAI.address,
+      "50",
+      daiUsdOracle.address,
+      []
+    );
+
+    bat = await daiFactory.deploy(); // Another erc20 token for uniswap router test
+    cBat = await cBatFactory.deploy(bat.address);
+    usdc = await (await ethers.getContractFactory("USDCMock")).deploy();
+    cUsdc = await (
+      await ethers.getContractFactory("cUSDCMock")
+    ).deploy(usdc.address);
+
+    await factory.createPair(bat.address, dai.address); // Create tokenA and dai pair
+    const pairAddress = factory.getPair(bat.address, dai.address);
+    await factory.createPair(bat.address, usdc.address);
+    usdcPair = new Contract(
+      await factory.getPair(bat.address, usdc.address),
+      JSON.stringify(IUniswapV2Pair.abi),
+      staker
+    );
+
+    pair = new Contract(
+      pairAddress,
+      JSON.stringify(IUniswapV2Pair.abi),
+      staker
+    ).connect(founder);
+    setDAOAddress("CDAI", cDAI.address);
+    setDAOAddress("DAI", dai.address);
+
+    //This set addresses should be another function because when we put this initialization of addresses in initializer then nameservice is not ready yet so no proper addresses
+    await goodReserve.setAddresses();
+    const gasFeeMockFactory = await ethers.getContractFactory(
+      "GasPriceMockOracle"
+    );
+    gasFeeOracle = await gasFeeMockFactory.deploy();
+    const daiEthPriceMockFactory = await ethers.getContractFactory(
+      "DaiEthPriceMockOracle"
+    );
+    daiEthOracle = await daiEthPriceMockFactory.deploy();
+
+    const ethUsdOracleFactory = await ethers.getContractFactory(
+      "EthUSDMockOracle"
+    );
+
+    batUsdOracle = await tokenUsdOracleFactory.deploy();
+    ethUsdOracle = await ethUsdOracleFactory.deploy();
+    await dai["mint(address,uint256)"](
+      founder.address,
+      ethers.utils.parseEther("10000000")
+    );
+    await bat["mint(address,uint256)"](
+      founder.address,
+      ethers.utils.parseEther("5001000")
+    );
+    await usdc["mint(address,uint256)"](
+      founder.address,
+      ethers.utils.parseUnits("10000000", 6)
+    );
+    await bat.transfer(cBat.address, ethers.utils.parseEther("1000000")); // We should put extra BAT to mock cBAT contract in order to provide interest
```

Comment:
is this required?

Reply:
Yeah for interest logic it's required that we should put extra token to ctoken in order to redeem tokens